### DISCUSSION TITLE
Kernel issue w/ Jupyter

### DISCUSSION THREAD

**Discussion by ChenxiSSS** [21cmfast/21cmFAST#270] [Q&A] [answered]
Hi! I am having an issue with the Jupyter kernel after the kernel is interrupted. Not sure if this is a Jupyter issue or 21cmFAST issue or maybe it is an issue from my server. Come to check if anyone could help :).

Running lightcone w/ Jupyterlab, I notice after the kernel is interrupted, sometimes manually or my recent unfortunately accidentally server reboot, re-run the notebook will not be successful. **The kernel of the notebooks will die wouldn't restart after the codes run for some time although given the notice `the kernel appears to have died. it will restart automatically`, and the behavior of the CPU load is different from those notebooks that execute the code successfully when inspecting the resources usage w/ `htop`.**

Or maybe there is a way to load back the cache from the disk after the simulation is interrupted halfway? I was getting close to finishing my long-running simulation before my last unfortunately accidentally server reboot due to air-conditioning issue @ the server room.

The 21cmFAST version 3.1.2 is installed using conda and managed using virtualenv.
Here are my params:
```python
HII_DIM = 1200
BOX_LEN = 300
N_THREADS = 24

user_params = {"HII_DIM":HII_DIM,  "BOX_LEN": BOX_LEN, "N_THREADS": N_THREADS}

lightcone = p21c.run_lightcone(
    redshift = 6.0,
    user_params = user_params,
    flag_options = flag_options,
    astro_params = astro_params,
    lightcone_quantities=("brightness_temp", 'density', 'xH_box'),
    global_quantities=("brightness_temp", 'density', 'xH_box'),
    direc='_cache'
)
```

**Comment by qyx268** [comment, marked as answer]
Could you please try examining the last file outputted? I was wondering if the kernel died when writing out the file. If so, then the next time when the code tries to reuse cache, it might crash when trying to read a corrupted file. 

Also, could you please try restarting without notebook? Just wanted to see if this is only happening with Jupyter.

**Reply by ChenxiSSS** [reply]
Hi Yuxiang! Thank you for the quick response. Unfortunately, my server had another due to air-conditioning issue after I post here. I will be trying to identify the last files as the server just came back online.

**Comment by ChenxiSSS** [comment]
Hi @qyx268, I moved the last file output by the kernel. It seems that now the kernel is not restarting, and the simulation seems to be fine. Many thanks for the help 😄 !

**Reply by ChenxiSSS** [reply]
Yeah, I can confirm that my simulation is all good and finished the rest redshift slices. 🥂
